src/instruments/src/PFD/SpeedIndicator.tsx

```diff
@@ -1,351 +1,919 @@
+            this.needsUpdate = true;
+        });
+
+        sub.on('targetSpeedManaged').whenChanged().handle((s) => {
+            this.speedState.managedTargetSpeed = s;
+            this.needsUpdate = true;
+        });
+
+        sub.on('autoBrakeActive').whenChanged().handle((a) => {
+            this.decelActive = a;
+            this.needsUpdate = true;
+        });
+
+        sub.on('realTime').handle(this.onFrameUpdate.bind(this));
+    }
+
+    private onFrameUpdate(_realTime: number): void {
+        if (this.needsUpdate === true) {
+            this.needsUpdate = false;
+
+            this.determineTargetSpeed();
+            const inRange = this.handleLowerUpperBound();
+            this.handleManagedSpeed();
+
+            if (inRange) {
+                const multiplier = 100;
+                const currentValueAtPrecision = Math.round(this.speedState.speed.value * multiplier) / multiplier;
+                const offset = (currentValueAtPrecision - (this.speedState.isSpeedManaged
+                    ? this.speedState.managedTargetSpeed : this.speedState.targetSpeed)) * DistanceSpacing / ValueSpacing;
+                this.speedTargetRef.instance.style.transform = `translate3d(0px, ${offset}px, 0px)`;
+            } else {
+                const text = Math.round(this.speedState.isSpeedManaged ? this.speedState.managedTargetSpeed : this.speedState.targetSpeed).toString().padStart(3, '0');
+                this.textSub.set(text);
+            }
+        }
+    }
+
+    private determineTargetSpeed() {
+        const isSelected = !this.speedState.isSpeedManaged;
+        if (isSelected) {
+            if (this.speedState.isMach) {
+                const holdValue = this.speedState.holdValue;
+                this.speedState.targetSpeed = SimVar.GetGameVarValue('FROM MACH TO KIAS', 'number', holdValue === null ? undefined : holdValue);
+            } else {
+                this.speedState.targetSpeed = this.speedState.holdValue;
+            }
+        }
+    }
+
+    private handleLowerUpperBound(): boolean {
+        let inRange = false;
+
+        const currentTargetSpeed = this.speedState.isSpeedManaged ? this.speedState.managedTargetSpeed : this.speedState.targetSpeed;
+        if (this.speedState.speed.value - currentTargetSpeed > DisplayRange) {
+            this.upperBoundRef.instance.style.visibility = 'visible';
+            this.lowerBoundRef.instance.style.visibility = 'hidden';
+            this.speedTargetRef.instance.style.visibility = 'hidden';
+            this.currentVisible = this.upperBoundRef;
+        } else if (this.speedState.speed.value - currentTargetSpeed < -DisplayRange && !this.decelActive) {
+            this.lowerBoundRef.instance.style.visibility = 'visible';
+            this.upperBoundRef.instance.style.visibility = 'hidden';
+            this.speedTargetRef.instance.style.visibility = 'hidden';
+            this.currentVisible = this.lowerBoundRef;
+        } else if (!this.decelActive) {
```

Comment:
The selected speed bug should not be always hidden when DECEL is active, instead, it should only show when it is
in range.
```suggestion
        } else if (Math.abs(airspeed - targetSpeed) < DisplayRange) {
```